src/main/java/server/Server.java

```diff
@@ -2,27 +2,32 @@
 
 import com.beust.jcommander.JCommander;
 import com.beust.jcommander.ParameterException;
+import main.java.application.App;
+import main.java.application.Routes;
 import main.java.server.CLI.Args;
 
 import java.io.IOException;
 import java.net.ServerSocket;
 import java.net.Socket;
 
 class Server {
-    private final Integer port;
+    private App app;
+    private Integer port;
     private boolean serverIsRunning = true;
 
-    private Server(Integer port) {
+    public Server(Integer port, App app) {
         this.port = port;
+        this.app = app;
```

Comment:
Awesome that you're injecting the App into the Server! I think it makes sense that the App would depend on the Server for defining interfaces that specify how to build a compatible app, and then the Server could be agnostic about any individual app's implementation details.